src/lib/utils/hooks/useGetInfoFromIds.ts

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useState } from 'react'
+import { EntityHeader, Reference, ReferenceList } from '../synapseTypes'
+import { getEntityHeader } from '../SynapseClient'
+import { getUserProfileWithProfilePicAttached } from '../functions/getUserData'
+import { UserProfile } from '../synapseTypes'
+import { SynapseConstants } from '..'
+import { without, chunk } from 'lodash-es'
+
+export type HookType = 'ENTITY_HEADER' | 'USER_PROFILE'
+export type UseGetInfoFromIdsProps = {
+  ids: string[]
+  token?: string
+  type: HookType
+}
+
+type LookupRequestType = string | Reference
+
+const UserProfileTemplate: UserProfile = {
+  ownerId: '', // A foreign key to the ID of the 'principal' object for the user
+  firstName: 'Unknown', // This person's given name (forename)
+  lastName: 'Unknown', // This person's family name (surname)
+  userName: 'Unknown', // A name chosen by the user that uniquely identifies them
+}
+
+const entityHeaderTemplate: EntityHeader = {
+  name: 'Unknown', //	The name of the entity
+  id: 'unknown', //	The id of the entity
+  type: 'unknown', //	The type of the entity
+  versionNumber: 0, //	The version number of the entity
+  versionLabel: 'placeholder', //	The user defined version label of the entity
+  benefactorId: 0, //	The ID of the entity that this Entity's ACL is inherited from.
+  createdOn: 'null', //	The date this entity was created.
+  modifiedOn: 'null', //	The date this entity was last modified.
+  createdBy: 'null', //	The ID of the user that created this entity.
+  modifiedBy: 'null', //	The ID of the user that last modified this entity.
+}
+
+const getEntityHeaderItems = async (
+  lookupList: ReferenceList,
+  token: string | undefined,
+): Promise<EntityHeader[]> => {
+  const newData = await getEntityHeader(lookupList as ReferenceList, token)
```

Comment:
is this cast `lookupList as ReferenceList` necessary?

Reply:
Good catch! I was moving the code around and forgot to remove. Thanks!